src/routes/consent.ts

```diff
@@ -1,60 +1,129 @@
 // SPDX-License-Identifier: Apache-2.0
-import { defaultConfig, RouteCreator, RouteRegistrator } from "../pkg"
+import { defaultConfig, logger, RouteCreator, RouteRegistrator } from "../pkg"
 import { register404Route } from "./404"
 import { oidcConformityMaybeFakeSession } from "./stub/oidc-cert"
-import {
-  AcceptOAuth2ConsentRequestSession,
-  IdentityApi,
-  OAuth2ConsentRequest,
-} from "@ory/client"
+import { AcceptOAuth2ConsentRequestSession } from "@ory/client"
 import { UserConsentCard } from "@ory/elements-markup"
 import bodyParser from "body-parser"
-import csrf from "csurf"
+import { doubleCsrf, DoubleCsrfCookieOptions } from "csrf-csrf"
+import { Request, Response, NextFunction } from "express"
 
-async function createOAuth2ConsentRequestSession(
-  grantScopes: string[],
-  consentRequest: OAuth2ConsentRequest,
-  identityApi: IdentityApi,
-): Promise<AcceptOAuth2ConsentRequestSession> {
-  // The session allows us to set session data for id and access tokens
+const cookieOptions: DoubleCsrfCookieOptions = {
+  sameSite: "lax",
+  signed: true,
+  // set secure cookies by default (recommended in production)
+  // can be disabled through DANGEROUSLY_DISABLE_SECURE_COOKIES=true env var
+  secure: true,
+  ...(process.env.DANGEROUSLY_DISABLE_SECURE_CSRF_COOKIES && {
+    secure: false,
+  }),
+}
 
-  const id_token: { [key: string]: any } = {}
+const cookieName = process.env.CSRF_COOKIE_NAME || "__Host-ax-x-csrf-token"
+const cookieSecret = process.env.CSRF_COOKIE_SECRET
 
-  if (consentRequest.subject && grantScopes.length > 0) {
-    const identity = (
-      await identityApi.getIdentity({ id: consentRequest.subject })
-    ).data
+// Sets up csrf protection
+const {
+  invalidCsrfTokenError,
+  doubleCsrfProtection, // This is the default CSRF protection middleware.
+} = doubleCsrf({
+  getSecret: () => cookieSecret || "", // A function that optionally takes the request and returns a secret
+  cookieName: cookieName, // The name of the cookie to be used, recommend using Host prefix.
+  cookieOptions,
+  ignoredMethods: ["GET", "HEAD", "OPTIONS", "PUT", "DELETE"], // A list of request methods that will not be protected.
+  getTokenFromRequest: (req: Request) => {
+    logger.debug("Getting CSRF token from request", { body: req.body })
+    return req.body._csrf
+  }, // A function that returns the token from the request
+})
 
-    if (grantScopes.indexOf("email") > -1) {
-      // Client may check email of user
-      id_token.email = identity.traits["email"] || ""
-    }
-    if (grantScopes.indexOf("phone") > -1) {
-      // Client may check phone number of user
-      id_token.phone = identity.traits["phone"] || ""
+// Checks if OAuth2 consent is enabled
+// This is used to determine if the consent route should be registered
+// We need to check if the environment variables are set
+const isOAuthCosentEnabled = () =>
+  (process.env.HYDRA_ADMIN_URL || process.env.ORY_SDK_URL) &&
+  process.env.CSRF_COOKIE_SECRET
```

Comment:
doesn't need to be a function.

Reply:
true. I'll change this. I also thought today of moving it inside the `createHelpers` function so that we can override this.